simulations/vip-178/vip-178/simulations.ts

```diff
@@ -0,0 +1,320 @@
+
+          before(async () => {
+            const vToken = await ethers.getContractAt(VTOKEN_ABI, vTokenAddress);
+            rateModel = await ethers.getContractAt(RATE_MODEL_ABI, await vToken.interestRateModel());
+          });
+
+          it(`should have base = ${base}`, async () => {
+            const basePerBlock = parseUnits(base, 18).div(BLOCKS_PER_YEAR);
+            expect(await rateModel.baseRatePerBlock()).to.equal(basePerBlock);
+          });
+
+          it(`should have jump = ${jump}`, async () => {
+            const jumpPerBlock = parseUnits(jump, 18).div(BLOCKS_PER_YEAR);
+            expect(await rateModel.jumpMultiplierPerBlock()).to.equal(jumpPerBlock);
+          });
+
+          it(`should have multiplier = ${multiplier}`, async () => {
+            const multiplierPerBlock = parseUnits(multiplier, 18).div(BLOCKS_PER_YEAR);
+            expect(await rateModel.multiplierPerBlock()).to.equal(multiplierPerBlock);
+          });
+
+          it(`should have kink = ${kink}`, async () => {
+            expect(await rateModel.kink()).to.equal(parseUnits(kink, 18));
+          });
+        });
+      };
+
+      describe("Interest rate models", () => {
+        checkInterestRate(VagEUR_Stablecoins, "VagEUR_Stablecoins", {
+          base: "0.02",
+          multiplier: "0.1",
+          jump: "2.5",
+          kink: "0.5",
+        });
+
+        describe("RF, CF, LT, Protocol seize share", () => {
+          it("should set VagEUR_Stablecoins reserve factor to 10%", async () => {
+            expect(await vagEUR.reserveFactorMantissa()).to.equal(parseUnits("0.10", 18));
+          });
+
+          it("should set VagEUR_Stablecoins collateral factor to 75%", async () => {
+            const market = await comptroller.markets(VagEUR_Stablecoins);
+            expect(market.collateralFactorMantissa).to.equal(parseUnits("0.75", 18));
+          });
+
+          it("should set VagEUR_Stablecoins liquidation threshold to 80%", async () => {
+            const market = await comptroller.markets(VagEUR_Stablecoins);
+            expect(market.liquidationThresholdMantissa).to.equal(parseUnits("0.80", 18));
+          });
+
+          it("should set VagEUR_Stablecoins protocolSeizeShareMantissa to 5%", async () => {
+            const protocolSeizeShare = await vagEUR.protocolSeizeShareMantissa();
+            expect(protocolSeizeShare).to.equal(parseUnits("0.05", 18));
+          });
+        });
+
+        describe("Caps", () => {
+          it("should set VagEUR_Stablecoins borrow cap to 50,000", async () => {
+            expect(await comptroller.borrowCaps(VagEUR_Stablecoins)).to.equal(parseUnits("50000", 18));
+          });
+
+          it("should set VagEUR_Stablecoins supply cap to 1,00,000", async () => {
+            expect(await comptroller.supplyCaps(VagEUR_Stablecoins)).to.equal(parseUnits("100000", 18));
+          });
+        });
+      });
+    });
+
+    describe("Reward Distributor", () => {
```

Comment:
We could use the check that we have here: https://github.com/VenusProtocol/vips/blob/develop/src/vip-framework/checks/rewardsDistributor.ts